### PR TITLE
2. update cluster creation instructions + exercise

### PR DESCRIPTION
**Pull request by wbuchwalter**

### CODE REVIEW

File: 2-kubernetes/README.md [hunk 1]
```diff
@@ -36,37 +36,36 @@ The basic Kubernetes objects include:
 * Service - an abstraction which defines a logical set of Pods and a policy by which to access them.
 * Volume - an abstraction which allows data to be preserved across container restarts and allows data to be shared between different containers.
 * Namespace - a way to divide a physical cluster resources into multiple virtual clusters between multiple users.
+* Deployment - Manages pods and ensures a certain number of them are running. This is tipically used to deploy pods that should always be up, such as a web server.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: typical

File: 2-kubernetes/README.md [hunk 2]
```diff
@@ -36,37 +36,36 @@ The basic Kubernetes objects include:
...
     metadata:
       labels:
         app: nginx
-    spec:
-      containers:
-      - name: nginx
-        image: nginx:1.7.9
-        ports:
+    spec: # Specification for the Pod 
+      containers: # This are the containers running inside our Pod, in our case a single one
```
Comment (ritazh) [inline 2-kubernetes/README.md]: These

File: 2-kubernetes/README.md [hunk 3]
```diff
@@ -36,37 +36,36 @@ The basic Kubernetes objects include:
...
         - containerPort: 80
 ```
 
-To create all the objects described in a Deployment using a `.yaml` file like the one above in your own Kubernetes cluster, run the following command using the `kubectl` command-line interface:
-
-```
-$ kubectl create -f https://k8s.io/docs/user-guide/nginx-deployment.yaml
-```
+To create all the objects described in a Deployment using a `.yaml` file like the one above in your own Kubernetes cluster you can use Kubernetes' CLI (`kubectl`). 
+We will try creating a deployment in the excercise toward the end of this module, but first we need a cluster.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: We will be creating

File: 2-kubernetes/README.md [hunk 4]
```diff
@@ -80,6 +79,16 @@ AKS is currently still in preview and acs-engine is a bit more complex to setup,
 We are going to create a Linux-based K8s cluster.
 You can either create the cluster using the portal, or using Azure-CLI (`az`).
 
+### A Note on GPUs with Kubernetes
+
+As of this writing, GPUs are still in preview with ACS.  
+You can deploy an ACS cluster with GPU VMs (such as `Standard_NC6`) in `uswest2` or `uksouth` but you should be aware of some pitfalls:
+* Deploying a GPU cluster takes longer than a CPU cluster (about 10-15 minutes more) because the NVIDIA drivers need to be installed as well.
+* Since this is a preview, you might hit capacity issues if the location you chose does not have enough GPUs available to accomodate you.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: accommodate

File: 2-kubernetes/README.md [hunk 5]
```diff
@@ -149,19 +157,122 @@ k8s-agent-ef2b999d-2    Ready     9d        v1.7.7
...
-Should indicate the presence of GPU(s) on the node:
-```
 [...]
 Capacity:
  alpha.kubernetes.io/nvidia-gpu:	1
 [...]
  ```
 
+## Excercise
+
```
Comment (ritazh) [inline 2-kubernetes/README.md]: exercise

File: 2-kubernetes/README.md [hunk 6]
```diff
@@ -149,19 +157,122 @@ k8s-agent-ef2b999d-2    Ready     9d        v1.7.7
...
+### Running our Model on Kubernetes
+
+> Note: If you didn't complete the exercise in module 1, you can use `wbuchwalter/tf-mnist` image for this excercise.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: exercise

File: 2-kubernetes/README.md (hunk 3 above)
Comment (ritazh) [reply 2-kubernetes/README.md]: exercise

File: 2-kubernetes/README.md [hunk 7]
```diff
@@ -80,6 +79,16 @@ AKS is currently still in preview and acs-engine is a bit more complex to setup,
...
 
+### A Note on GPUs with Kubernetes
+
+As of this writing, GPUs are still in preview with ACS.  
+You can deploy an ACS cluster with GPU VMs (such as `Standard_NC6`) in `uswest2` or `uksouth` but you should be aware of some pitfalls:
+* Deploying a GPU cluster takes longer than a CPU cluster (about 10-15 minutes more) because the NVIDIA drivers need to be installed as well.
+* Since this is a preview, you might hit capacity issues if the location you chose does not have enough GPUs available to accomodate you.
+
+**Unless you are already pretty familiar with docker and Kubernetes, we recommend that you create a cluster with CPU VMs to save some time.**
+Only module 3 has an excercise which is specific for GPU VMs, all other modules can be followed on either CPU or GPU clusters.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: exercise

File: 2-kubernetes/README.md [hunk 8]
```diff
@@ -106,7 +114,7 @@ With:
   
 | Parameter | Description |
 | --- | --- | 
-| AGENT_SIZE | The size of K8s's agent VM. For this workshop use `Standard_NC6` if you want GPUs or `Standard_D2_v2` for CPU only. |
+| AGENT_SIZE | The size of K8s's agent VM. `Standard_D2_v2` is enough for this workshop. |
```
Comment (ritazh) [inline 2-kubernetes/README.md]: shouldn't we mention 
> use `Standard_NC6` if you want GPUs

File: 2-kubernetes/README.md [hunk 9]
```diff
@@ -149,19 +157,122 @@ k8s-agent-ef2b999d-2    Ready     9d        v1.7.7
...
+
+### Running our Model on Kubernetes
+
+> Note: If you didn't complete the exercise in module 1, you can use `wbuchwalter/tf-mnist` image for this excercise.
+
+In module 1, we created an image for our MNIST classifier, ran a small training locally and pushed this image to Docker Hub.  
+Since we now have a running Kubernetes cluster, let's run our training on it!
+
+First, we need to create a YAML template to define what we want to deploy.
+We want our deployment to have a few caracteristics:
```
Comment (ritazh) [inline 2-kubernetes/README.md]: characteristics

File: 2-kubernetes/README.md (hunk 8 above)
Comment (wbuchwalter) [reply 2-kubernetes/README.md]: I want to avoid people creating GPU clusters for now, unless they really know what they are doing.

Comment (ritazh) [reply 2-kubernetes/README.md]: ok

**Pull request merged**